apps/s_server.c

```diff
@@ -2594,8 +2594,7 @@ static int sv_body(int s, int stype, int prot, unsigned char *context)
                     continue;
                 }
                 if (buf[0] == 'P') {
-                    static const char *str = "Lets print some clear text\n";
-                    BIO_write(SSL_get_wbio(con), str, strlen(str));
+                    BIO_write(SSL_get_wbio(con), "Lets print some clear text\n", sizeof("Lets print some clear text\n") -1);
```

Comment:
Make the text string a "static const char text[] = " and then use sizeof on that.

Reply:
> The declaration static const char text[] is complete unnecessary.

No @raniervf, it is not unnecessary. It avoids duplicating the message as you did. @richsalz's suggestion was actually to change the `const char *` into a `const char []`, which gives a much nicer result:

```patch
index 5fa7c2fb42..08e73202ea 100644
--- a/apps/s_server.c
+++ b/apps/s_server.c
@@ -2594,8 +2594,8 @@ static int sv_body(int s, int stype, int prot, unsigned char *context)
                     continue;
                 }
                 if (buf[0] == 'P') {
-                    static const char *str = "Lets print some clear text\n";
-                    BIO_write(SSL_get_wbio(con), str, strlen(str));
+                    static const char str[] = "Lets print some clear text\n";
+                    BIO_write(SSL_get_wbio(con), str, sizeof(str)-1);
                 }
                 if (buf[0] == 'S') {
                     print_stats(bio_s_out, SSL_get_SSL_CTX(con));
```

If the difference between the `const char *` and the `const char []` is not obvious to you, the following toy example might help to understand it ;-)


**testo.c**
```C
#include <stdio.h>

static const char* text1   = "Lets print some clear text\n";
static const char  text2[] = "Lets print some clear text\n";

void main(int argc, char* argv[])
{
    printf("sizeof(text1) = %lu\n", sizeof(text1));
    printf("sizeof(text2) = %lu\n", sizeof(text2));
}
```

```bash
$ make testo

$ ./testo 
sizeof(text1) = 8
sizeof(text2) = 28
```

Reply:
It seems he meant the `static` is not necessary

Reply:
In my opinion the `static` is necessary, because it it is static const then the compiler does not have to copy the characters onto the stack on every call. (He might be smart enough to do it anyway, but it's better to be explicit.)